core/src/main/scala/org/apache/spark/deploy/worker/Worker.scala

```diff
@@ -185,6 +185,7 @@ private[deploy] class Worker(
 
   var coresUsed = 0
   var memoryUsed = 0
+  val resourcesUsed = new HashMap[String, ResourceInformation]()
```

Comment:
Yeah I don't think you should use REsourceInformation here to track used since like you said its immutable. We don't want to change that to be mutable because its user facing API.  Something more like WorkerResourceInfo would make sense

Reply:
Oh, thank you @tgravescs for reminding this. I'll update it as you suggested.